src/controllers/profileController.ts

```diff
@@ -95,55 +54,35 @@ export default class ProfileController {
   public upsertProfile(req: express.Request, res: express.Response): void {
     const maybeUser = extractUserFromRequest(req);
 
-    maybeUser.fold(
-      (error: string) => {
+    if (isLeft(maybeUser)) {
+      // Unable to extract the user from the request.
+      const error = maybeUser.value;
+      res.status(500).json({
+        message: error
+      });
+      return;
+    }
+
+    const maybeUpsertProfile = extractUpsertProfileFromRequest(req);
+
+    if (isLeft(maybeUpsertProfile)) {
+      // Unable to extract the upsert profile from the request.
+      const error = maybeUser.value;
+      res.status(500).json({
+        message: error
+      });
+      return;
+    }
+
+    this.profileService
+      .upsertProfile(maybeUser.value, maybeUpsertProfile.value)
+      .then(data => {
+        res.json(data);
```

Comment:
you can plug some quick&dirty type checking here, if you return the Promise instead of void ie. 

```
public upsertProfile(req: express.Request, res: express.Response): Promise<Either<Error, Profile>> {
...
 return this.profileService.upsertProfile(maybeUser.value, maybeUpsertProfile.value)
    .then(data => { res.json(data); return right(data); })
}
```

probably useful for testing. 
@cloudify  ?

Reply:
for testing we can use a `Response` mock, I would not add more logic to this code at this moment, as we may refactor it to separate the business logic from the request handling